src/main/java/org/folio/rest/support/EffectiveCallNumberComponentsUtil.java

```diff
@@ -0,0 +1,44 @@
+package org.folio.rest.support;
+
+import static org.apache.commons.lang.StringUtils.isNotBlank;
+
+import org.folio.rest.jaxrs.model.EffectiveCallNumberComponents;
+import org.folio.rest.jaxrs.model.HoldingsRecord;
+import org.folio.rest.jaxrs.model.Item;
+
+public final class EffectiveCallNumberComponentsUtil {
+
+  public static EffectiveCallNumberComponents buildComponents(HoldingsRecord holdings, Item item) {
+    return updateComponents(new EffectiveCallNumberComponents(), holdings, item);
+  }
+
+  private static EffectiveCallNumberComponents updateComponents(
+    EffectiveCallNumberComponents components, HoldingsRecord holdings, Item item) {
+
+    String updatedCallNumber = null;
```

Comment:
Maybe this instead:
```suggestion
    String updatedCallNumber = StringUtils.firstNonBlank(item.getItemLevelCallNumber(), holdings.getCallNumber());
```

Reply:
It seems that `holdings` will be null here if `shouldNotRetrieveHoldings` method returns `true`, so possible NPE here.

Reply:
If we do this we will just end up replacing the string logic with unwrapping the optional. It seems to be pretty equivalent to me.

Reply:
Good point. Though, I think using `Optional` adds some complexity. Instead of sending `null`, we could send `new HoldingsRecord()`. Nothing would need to change in this code. Since the public static method is a utility method, we may want to add `Objects.requireNonNull()` checks for the holdings and item arguments to indicate to a user of this utility class that they cannot send `null` objects to the public method.

Reply:
I haven't done something like this is in the ItemStorageAPI.java because I'm passing null for the holdings after explicitly checking that all the item properties exist to avoid fetching the holdings unnecessarily.

Reply:
Right, but to Bohdan's point, when we call `holdings.getCallNumber()` below we'll get a NPE since `holdings` is `null`. So, my thought is that in `ItemStorageAPI`, instead of calling:
```java
     item.setEffectiveCallNumberComponents(EffectiveCallNumberComponentsUtil.buildComponents(null, item));
```
we could do something like:
```java
     item.setEffectiveCallNumberComponents(EffectiveCallNumberComponentsUtil.buildComponents(new HoldingsRecord(), item));
```
Then `holdings.getCallNumber()` won't NPE and will return `null` since the `HoldingsRecord` has nothing set. We already have checked that we don't need holdings info. It may be a bit of a hack, but I've seen worse. We could create a constant `EMPTY_HOLDINGS_RECORD` or something like that to pass in if it seems strange to create a new `HoldingsRecord` after determining that we don't need one. Another option is another static method here without the holdings record argument and pass in a newly constructed `HoldingsRecord`. At least that takes some of the confusion out of the code using the utility.
```java
  public static EffectiveCallNumberComponents buildComponents(Item item) {
    return updateComponents(new EffectiveCallNumberComponents(), new HoldingsRecord(), item);
  }
```